plugins/catalog/src/js/repositories/repositoriesStream.js

```diff
@@ -0,0 +1,32 @@
+import { request } from "@dcos/http-service";
+import Config from "#SRC/js/config/Config";
+
+export const fetchRepositories = type =>
+  request(`${Config.rootUrl}${Config.cosmosAPIPrefix}/repository/list`, {
+    method: "POST",
+    body: JSON.stringify({ type }),
```

Comment:
What do you think about making the request function support object bodies and automatically stringify?

Reply:
The idea behind http-service was that it is rather dumb. Having `request` stringifying payloads is a bit more magic

Reply:
totally fair. I was just thinking the most likely case is that we are passing a json body that needs to be stringified, having to do this as the consumer of the api every time seemed less than ideal.  I agree too much magic is no good, *if* we were going to do something, maybe adding a `json` property that tells request to stringify the object, would be a compromise, but not saying its necessary.